presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java

```diff
@@ -153,19 +153,19 @@ public void completionEvent(QueryInfo queryInfo)
         }
     }
 
-    private static Optional<TaskInfo> findFailedTask(StageInfo stageInfo)
+    private static Optional<TaskInfo> findFailedTask(Optional<StageInfo> stageInfo)
```

Comment:
Good idea. Fixed as suggested.

On Tue, Mar 8, 2016 at 11:49 PM, Martin Traverso <notifications@github.com>
wrote:

> In
> presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java
> <https://github.com/facebook/presto/pull/4724#discussion_r55481737>:
>
> > @@ -153,19 +153,19 @@ public void completionEvent(QueryInfo queryInfo)
> >          }
> >      }
> >
> > -    private static Optional<TaskInfo> findFailedTask(StageInfo stageInfo)
> > +    private static Optional<TaskInfo> findFailedTask(Optional<StageInfo> stageInfo)
>
> Instead of making this take an Optional (which is a bit unnatural), make
> it take a StageInfo and call this method above like this:
>
>  Optional<TaskInfo> task = queryInfo.getOutputStage().map(QueryMonitor::findFailedTask);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/presto/pull/4724/files#r55481737>.
>